src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6740,8 +6819,79 @@ private static bool ReportRefOrOutArgument(AnalyzedArguments analyzedArguments,
             return false;
         }
 
+        /// <summary>
+        /// PROTOTYPE: remove this method before shipping (this is for demo purposes only)
+        /// </summary>
+        private BoundExpression GetPrototypeExtensionIndexerOpt(ExpressionSyntax syntax, BoundExpression receiver, AnalyzedArguments analyzedArguments, DiagnosticBag diagnostics)
+        {
+            // can only have one unnamed argument, with no ref kind.
+            if (analyzedArguments.Arguments.Count != 1
+                || analyzedArguments.Argument(0).Type is null
+                || analyzedArguments.RefKinds.Any()
+                || analyzedArguments.Names.Any())
+            {
+                return null;
+            }
+
+            BoundExpression argument = analyzedArguments.Argument(0);
+
+            // argument must be a range or an index
+            switch (argument.Type.ToDisplayString())
+            {
+                case "System.Index":
+                case "System.Range":
+                    break;
+                default:
+                    return null;
+            }
+
+            // receiver must be a string, a span, or an array
+            if (receiver.Type.ToDisplayString() != "string"
```

Comment:
>ToDisplayString [](start = 30, length = 15)

This is very wrong again. We have IsStringType helper to perform this check reliably.